bundles/org.openhab.core.io.rest.voice/src/main/java/org/openhab/core/io/rest/voice/internal/VoiceResource.java

```diff
@@ -229,8 +230,14 @@ public Response getDefaultVoice() {
             @ApiResponse(responseCode = "200", description = "OK") })
     public Response say(@Parameter(description = "text to speak", required = true) String text,
             @QueryParam("voiceid") @Parameter(description = "voice id") @Nullable String voiceId,
-            @QueryParam("sinkid") @Parameter(description = "audio sink id") @Nullable String sinkId) {
-        voiceManager.say(text, voiceId, sinkId);
+            @QueryParam("sinkid") @Parameter(description = "audio sink id") @Nullable String sinkId,
+            @QueryParam("volume") @Parameter(description = "volume level") @Nullable String volume,
+            @QueryParam("enableCache") @Parameter(description = "enable TTS cache") Boolean enableCache) {
+        PercentType volumePercent = null;
+        if (volume != null && !volume.isEmpty()) {
+            volumePercent = new PercentType(volume);
+        }
+        voiceManager.say(text, voiceId, sinkId, volumePercent, enableCache);
```

Comment:
@lolodomo provides me with a use case for bypassing the cache : 
https://github.com/openhab/openhab-core/issues/3039#issuecomment-1210343492
But simplifying the code and parameters here is also a good objective. I have to say I was not fond of adding the numerous methods for this parameter 😅
What shoud I do ?

Reply:
The contribution guidelines say:

> We're trying very hard to keep openHAB lean and focused. We don't want it to do everything for everybody.

I don't think we should blow up API to cover every use-case. At maximum this will add one cache-miss (because the oldest entry is removed and would be a miss on next call). Of course the rate increases the more one-time calls you make, but on the other hand: if the majority of your calls is not to be cached, then using the cache makes no sense anyway.

Reply:
OK, I'm convinced. And the added complexity of a new parameter is a strong argument to my lazy part.
@lolodomo do you have a counter argument, or is it also OK for you ?